test/unit/ngsiv2/mqttBinding-test.js

```diff
@@ -0,0 +1,346 @@
+            });
+        });
+    });
+
+    describe('When a new measure arrives for an unprovisioned Device', function() {
+        var groupCreation = {
+                url: 'http://localhost:4041/iot/services',
+                method: 'POST',
+                json: utils.readExampleFile('./test/unit/ngsiv2/groupProvisioning/provisionFullGroup.json'),
+                headers: {
+                    'fiware-service': 'TestService',
+                    'fiware-servicepath': '/testingPath'
+                }
+            };
+
+        beforeEach(function(done) {
+            // Note: /v1/updateContext response is not processed by IOTA so its content is irrelevant,
+            // as far as it is a 200 OK
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v1/updateContext')
+                .reply(200, '{}');
+
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/entities/SensorMachine:UL_UNPROVISIONED/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/unprovisionedMeasure.json'))
+                .reply(204);
+
+            request(groupCreation, function(error, response, body) {
+                done();
+            });
+        });
+
+        it('should send a new update context request to the Context Broker with just that attribute', function(done) {
+            mqttClient.publish('/80K09H324HV8732/UL_UNPROVISIONED/attrs/temperature', '23', null, function(error) {
+                setTimeout(function() {
+                    contextBrokerMock.done();
+                    done();
+                }, 100);
+            });
+        });
+    });
+
+
+    describe('When a new multiple measure arrives to a Device topic with one measure', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send a single update context request with all the attributes', function(done) {
+            mqttClient.publish('/1234/MQTT_2/attrs', 'temperature|23', null, function(error) {
+                setTimeout(function() {
+                    contextBrokerMock.done();
+                    done();
+                }, 100);
+            });
+        });
+    });
+
+
+    describe('When a new multiple measure arrives to a Device topic with a faulty payload', function() {
+        beforeEach(function() {
+            contextBrokerMock
```

Comment:
In the case of faulty measure I understand nothing is sent to CB. Thus, no need to mock.

Reply:
Seems to be a similar case to https://github.com/telefonicaid/iotagent-ul/pull/242#discussion_r170203590.

As far as I understand, in that case mock code was finally removed, so I undesrtand we can do the same in this case.

Reply:
Fixed in bbf86fa4daf189c3fd03c3e5477673ca8aa25157

Reply:
The mock is still there... Maybe I am wrong? Or the removal is pending (although I see a "Fixed in" comment above)